src/m_pd.h

```diff
@@ -844,6 +844,7 @@ typedef union
 
 static inline int PD_BADFLOAT(t_float f)  /* malformed float */
 {
+    if (f == 0) return 0;
```

Comment:
small nitpick: Pd is still nominally a C89 codebase, so all declarations need to be at the top of the block. Just move the statement down one line and you're good. Same for the double precision version below.